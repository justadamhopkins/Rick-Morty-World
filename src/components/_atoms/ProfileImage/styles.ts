import styled from "styled-components";
import { breakpoint } from "_styles/BreakpointStyles";

export const StyledPortfolioWrapper = styled.div`
  max-width: 250px;
  max-height: 250px;
  border-radius: 1000px;
  overflow: hidden;
  vertical-align: middle;
  border: 7px solid var(--secondaryBackground);
  width: 100%;
  margin: 10px 0 20px;
`;
