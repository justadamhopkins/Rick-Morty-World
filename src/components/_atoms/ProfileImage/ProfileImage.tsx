import { FC } from "react";
import { StyledPortfolioWrapper } from "./styles";
import Image from "next/image";

interface Props {
  image: string;
  alt: string;
}

export const ProfileImage: FC<Props> = ({ image, alt }) => {
  return (
    <StyledPortfolioWrapper>
      <Image
        src={image}
        alt={alt}
        height={400}
        width={400}
        layout="responsive"
      />
    </StyledPortfolioWrapper>
  );
};

export default ProfileImage;
