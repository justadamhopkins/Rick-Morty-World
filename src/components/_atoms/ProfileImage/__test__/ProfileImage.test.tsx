import { render } from "@testing-library/react";
import { ProfileImage } from "../ProfileImage";

describe("ProfileImage", () => {
  const renderProfileImage = () => {
    return render(<ProfileImage image="mockImage" alt="mockImageName" />);
  };
  test("should render a ProfileImage", () => {
    const { asFragment } = renderProfileImage();

    expect(asFragment()).toMatchSnapshot();
  });
});
