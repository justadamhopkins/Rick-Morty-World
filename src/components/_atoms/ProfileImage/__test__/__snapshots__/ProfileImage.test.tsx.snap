// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImage should render a ProfileImage 1`] = `
<DocumentFragment>
  <div
    class="styles__StyledPortfolioWrapper-abrk26-0 krggJI"
  >
    <div
      style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="display: block; box-sizing: border-box; padding-top: 100%;"
      />
      <img
        alt="mockImageName"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
